### PR TITLE
RootNode possible before calling the destructor has been deleted

### PR DESCRIPTION
**Pull request by HoGarfield**

### CODE REVIEW

**Comment by zifter** [comment]
Good PR, thanks!

File: cocos/editor-support/cocosbuilder/CCBAnimationManager.cpp [hunk 1]
```diff
@@ -99,7 +99,11 @@ Node* CCBAnimationManager::getRootNode()
 
 void CCBAnimationManager::setRootNode(Node *pRootNode)
 {
+    CC_SAFE_RELEASE_NULL(_rootNode);
+    
     _rootNode = pRootNode;
+    
+    CC_SAFE_RETAIN(_rootNode);
```
Comment (zilongshanren) [inline cocos/editor-support/cocosbuilder/CCBAnimationManager.cpp:106]: You retain the rootNode here, I think it should be release somewhere, like the destructor method, right?

Comment (HoGarfield) [inline cocos/editor-support/cocosbuilder/CCBAnimationManager.cpp:106]: Yes, you're right

Comment (zilongshanren) [inline cocos/editor-support/cocosbuilder/CCBAnimationManager.cpp:106]: @HoGarfield  So could you add it in this PR?
 After it CI is ready, we could merge this patch.
Thanks.

Comment (HoGarfield) [inline cocos/editor-support/cocosbuilder/CCBAnimationManager.cpp:106]: @andyque OK,but I can not add it now. I need some time to download the project and I need to wait until after work to do

Comment (zilongshanren) [inline cocos/editor-support/cocosbuilder/CCBAnimationManager.cpp:106]: @HoGarfield  I think you should also add `CC_SAFE_RELEASE_NULL(_rootNode);` in CCBAnimationManager's destructor.

**Comment by minggo** [comment]
Any update of this PR?

**Pull request closed without merging**